File: 1490/clone-n-ary-tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children if children is not None else []
"""

class Solution:
    def cloneTree(self, root: 'Node') -> 'Node':
        if root==None: # 空指標另外處理
            return None
        ans = Node(root.val)
        for child in root.children:
            one = self.cloneTree(child)
            ans.children.append(one)

        return ans

File: 1490/clone-n-ary-tree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    Node* cloneTree(Node* root) {
        if(root==nullptr) return nullptr;
        
        Node* ans = new Node(root->val);
        vector<Node*> children;
        for(Node* child : root->children){
            children.push_back(cloneTree(child));
        }
        ans->children = children;
        return ans;
    }
};
//case 2/38: []
